src/components/date-picker/date-picker.tsx

```diff
@@ -195,26 +135,63 @@ export class DatePicker {
                     helperText={this.helperText}
                     required={this.required}
                     value={this.formattedValue}
-                    onChange={this.handleChange}
+                    onFocus={this.showPortalElement}
+                    ref={(el) => (this.textField = el)}
                 />
+                <limel-portal
+                    containerId={this.portalId}
+                    visible={this.showPortal}
+                >
+                    <limel-flatpickr-adapter
+                        type={this.type}
+                        value={this.value}
+                        ref={(el) => (this.datePickerCalendar = el)}
+                        onChange={this.handleCalendarChange}
+                    />
+                </limel-portal>
             </div>
         );
     }
 
-    @Listen('resize', { target: 'window' })
-    public resizeEvent() {
-        this.picker.init(this.input, this.container, this.value);
-    }
-
     @Watch('value')
     protected onValueChange(newValue, oldValue) {
-        if (newValue !== oldValue) {
-            this.formattedValue = this.picker.formatDate(newValue);
+        if (newValue !== oldValue && newValue !== this.formattedValue) {
+            this.formattedValue = this.dateFormatter.formatDateByType(
+                newValue,
+                this.type
+            );
         }
     }
 
-    private handleChange(event) {
+    private showPortalElement(event) {
+        this.showPortal = true;
+        const inputElement = this.textField.shadowRoot.querySelector('input');
+        setTimeout(() => {
+            this.datePickerCalendar.inputElement = inputElement;
+        });
         event.stopPropagation();
-        this.formattedValue = event.detail;
+
+        document.addEventListener('mousedown', this.eventListener, { passive: true });
```

Comment:
The menu doesn't close when tabbing out of the input, so we probably need a listener for that too.

Reply:
This seems to be the code that handles this in flatpickr: https://github.com/flatpickr/flatpickr/blob/00036f843b2882fbf810b8e4639a02c34a28ad7d/src/index.ts#L1427

What event do you think we need to listen to for that to be fixed?

Reply:
They bind that handler to `touchstart`, `click`, and `focus` here: https://github.com/flatpickr/flatpickr/blob/00036f843b2882fbf810b8e4639a02c34a28ad7d/src/index.ts#L427